challenges/fullstack/challenge.md

```diff
@@ -19,148 +14,124 @@ We would like to evaluate your skill in the following areas:
+
+## Requirements - Level 4
 
 Implement an application that allows a user to browse directory content on a
 remote server.
 
 This application should have the following functionality:
 
-* A directory viewer client with client-side filtering and sorting capabilities
-  (level 1) and url-based navigation (level 2).
-* A server for serving the webassets, containerized using Docker (level 2).
-* Authentication with a login screen and backend API to browse a directory
-  (level 3+).
-
-### Level 1
-
-This level focuses on JS/CSS/DOM knowledge and does not require any server-side
-knowledge to implement.
-
-Create a directory viewer with the following functionality:
-
-* An interface for viewing and browsing directory content.
-* Filtering on filename within the current directory (basic string matching).
-* Sorting on filename, type, and size within the current directory.
-* Unlike simlar interfaces such as Google Drive and Dropbox, **file preview is
-  _not_ required**. Clicking on a file should be a no-op.
-
-Directions:
-
-*  Build your own components.
-*  Display the `name`, `size`, and `type` file attributes.
-*  Use human readable format to display the `size` of a file.
-*  Use below JSON structure to mock real directory.
-
-```js
-const directory = {
-  name: "teleport",
-  size: 0, // bytes
-  type: "dir",
-  items: [
-    {
-      name: "lib",
-      size: 0, // bytes
-      type: "dir",
-      items: [
-        {
-          name: "teleport.go",
-          size: 320000, // bytes
-          type: "file",
-        },
-        {
-          name: "test.go",
-          size: 3320000, // bytes
-          type: "file",
-        },
-      ],
-    },
-    {
-      name: "README.md",
-      size: 4340000, // bytes
-      type: "file",
-    },
-  ],
-};
-```
-
-You can use a design of similar viewers like Google Drive or Github as a
-starting point and then add missing pieces to it. For example, you can borrow
-CSS from these design systems:
-[Github](https://primer.style/components/getting-started) or
-[Google](https://material-ui.com/getting-started/usage/) if you like.
+* A Go backend that serves the webapp and an API
+* The UI, which should include client-side filtering and sorting capabilities
+  and URL-based navigation.
+* Strong authentication
 
-<img src="./assets/google.jpg" height="300" />
-<img src="./assets/github.jpg" height="200" />
+### API
 
-### Level 2
+This repository includes just enough starter code to serve up both the web app
+and a sample API endpoint.
 
-This level focuses on serving web assets. In addition to the Level 1
-requirements, you will:
+The starter code here only listens for plain-text HTTP. At Teleport, we avoid
+plain-text connections and prefer to encrypt all data in transit. You will be
+responsible for adding TLS.
 
-* Serve web assets from a web server.
-* Containerize your application using Docker.
-* Implement URL Navigation
+Your API only needs to return information about a single directory. There is no
```

Comment:
Right now it sounds like the API always returns one static directory. Might want to make it clear that the user can request a directory listing for any directory, but the content listing for that directory only needs to be contents of that directory and not recursive listing.

To be honest, typing this out in a clear way was hard for me, so maybe an example would help here?